src/plugins/integrations/HttpIntegration.ts

```diff
@@ -115,16 +115,30 @@ class HttpIntegration implements Integration {
 
           const req = request.call(this, options, wrappedCallback);
 
-          req.on('socket', () => {
-            if (req.listenerCount('response') === 1) {
-              req.on('response', (res: any) => res.resume());
-            }
-          });
-
           req.on('response', (res: any) => {
             span.setTag(HttpTags.HTTP_STATUS, res.statusCode);
           });
 
+          const emit = req.emit;
+          req.emit = function (eventName: any, arg: any) {
+            if (eventName === 'socket') {
+              if (req.listenerCount('response') === 1) {
+                req.on('response', (res: any) => res.resume());
+              }
+
+              if (!config.maskHttpBody && arg._httpMessage._hasBody) {
+                 try {
+                  const lines = arg._httpMessage.output[0].split('\n');
+                  span.setTag(HttpTags.BODY, lines[lines.length - 1]);
```

Comment:
Can't we get request body from `options`?
I mean `const body = options.body || ''`

Reply:
Nope. See the docs for http request options docs :  https://nodejs.org/api/http.html#http_http_request_options_callback

Http request options do not have `body` parameter, below code is a typical http post request.
As you can see you write the post data to request object, so I can only extract the post data by listening `socket` event.  `options.body ` does not work. This is the cleanest way, I came up with to request body.

```
const postData =JSON.stringify({
  'msg': 'Hello World!'
});

const options = {
  hostname: 'www.google.com',
  port: 80,
  path: '/upload',
  method: 'POST',
  headers: {
            'Authorization': `Token ${TOKEN}`,
            'Content-Type': 'application/json'
      }
};

const req = http.request(options, (res) => {
});

// Write data to request body
req.write(postData);
req.end();
```

Reply:
How it behaves if body is multi-line? As far as I see, we get the latest line